lib/bank_account/account.rb

```diff
@@ -0,0 +1,26 @@
+module Bank
+  class Account
+    attr_accessor :id, :balance
+
+    def initialize(id, initial_balance)
+      # raises error for negative initial balance
+      raise ArgumentError.new "Invalid initial balance" unless initial_balance > 0
+      @id = id
+      @balance = initial_balance
+    end
+
+    # subtracts amount from total balance
+    def withdraw(amount)
+      @balance -= amount
+      # for withdrawal greater than balance
+      # gives warning error and returns original balance
+      if @balance < 0
+        puts "Insufficient funds. Your remaining balance is:"
+        return @balance += amount
+      else
+        return @balance
+      end
+    end
```

Comment:
This totally works, but you might want to think about checking if there's enough remaining in `@balance` before subtracting, just in case anything goes haywire before you can add the withdrawn amount back in.